cmd/create/operatorroles/by_prefix.go

```diff
@@ -522,29 +522,51 @@ func buildCommandsFromPrefix(r *rosa.Runtime, env string,
 			Build()
 
 		var attachSharedVpcRolePolicy string
-		var createPolicyCommand string
+		var policyCommands []string
 
 		if isSharedVpc { // HCP Shared VPC policy attachment
-			var policyName string
-			if credrequest == aws.IngressOperatorCloudCredentialsRoleType {
-				createPolicyCommand, policyName = getHcpSharedVpcPolicyDetails(r, sharedVpcRoleArn, roleName,
-					iamTags, path)
-			} else if credrequest == aws.ControlPlaneCloudCredentialsRoleType {
-				createPolicyCommand, policyName = getHcpSharedVpcPolicyDetails(r, vpcEndpointRoleArn, roleName,
-					iamTags, path)
+
+			// Precreate HCP shared VPC policies for less memory usage + time to execute
+			var policyDetails = make(map[string][]string)
+			createPolicyCommand, policyName, err := getHcpSharedVpcPolicyDetails(r, sharedVpcRoleArn,
+				iamTags)
+			if err != nil {
+				return "", err
+			}
+			policyDetails[aws.IngressOperatorCloudCredentialsRoleType] = []string{createPolicyCommand, policyName}
```

Comment:
nit: really don't like using arbitrary structures like that. Would you mind creating some transition struct instead?
Like 
```
type manualPolicyData struct {
command string
name string
}
```